src/backend/pipeline/pipeline.py

```diff
@@ -38,9 +58,14 @@ def run(self, imgs: Mat | list[Mat]) -> Data:
         Returns:
             The processed data.
         """
+        # start time of the pipeline
+        start = time.time()
```

Comment:
I looked at the Firebase table, and you're saving the date as a number. It should be a `timestamp` (with timezone). As far as I can tell the format is `DD/MM/YYYY, HH:MM:SS`, not sure about the timezone.

Reply:
I cannot make this work for some reason. firestore.SERVER_TIMESTAMP throws an error once you first write to Firebase.